Upgrade to nixpkgs-master

proger opened the pull request:
- also squeezes new upcast

proger committed:
use shellCheck from nixpkgs

proger committed:
bump all the things

proger committed:
allow backporting hackage databases, allows sproxy to build

this hackage-packages.nix is hand-edited file that comes from
nixpkgs 0d919b374c1611f5e5cc082577111dba797f2e4d~1

proger committed:
myrapi wants old servant

proger committed:
ensure ghc 7.8.4 is the default

proger committed:
remove mariadb-galera

proger committed:
sync with new nixos (stub more things, removed ec2 blackholing)

proger committed:
drop old ruby stuff

proger committed:
pivotal_agent: sync with new ruby, turn into a package

proger committed:
php: update buildInputs and use multilib mysql

proger committed:
more eval fixes

proger committed:
properly stub virtualbox

proger committed:
stdenv.gcc -> stdenv.cc

proger committed:
reimplement compileHaskell using ghcWithPackages

proger committed:
remove galera-wsrep from sdk

phunehehe commented:
you probably know it already, but https://github.com/NixOS/nixpkgs/tree/release-15.09 is now usable

proger committed:
sdk: override linux, make nixos use nixpkgs-config

proger committed:
sdk: remove lots of things from qemu

proger committed:
nixos: stub connman, include generic loader module

proger committed:
mariadb: disable hardcoded paths

proger committed:
mkebs: hardcoding a region is bad mmkay

proger committed:
mkebs: capital AZ

proger committed:
amazon-image: do not install grub for hvm, install grub earlier in mkebs

proger committed:
fixed print-host-key

proger committed:
port unicron to new haskellng; disable tests

proger committed:
upcast: do not remove share/

proger committed:
bump upcast

proger committed:
sdk: add ShellCheck

proger committed:
these are not the droids you're looking for

proger commented on pkgs/default.nix:
```diff
@@ -23,34 +29,23 @@ let
     # Take a Haskell file together with its dependencies, produce a binary.
-    # Note: dependencies that are included in GHC itself must not be provided,
-    # as that would result in empty elements in GHC_PACKAGE_PATH. FIXME maybe.
     compileHaskell = deps: file:
```
should this function take a list of strings to avoid passing wrong libraries?..

proger committed:
compileHaskell: use mv

% ssh ci.zalora.net nix-store -l /nix/store/9wlz85dw5ckmznv3ma8jar7ig3c5yagd-Main.hs-compiled.drv                     ~
[1 of 1] Compiling Main             ( /nix/store/hkb0q7w88jkwhmq3lv0q0m0q6agcn612-Main.hs, /nix/store/hkb0q7w88jkwhmq3lv0q0m0q6agcn612-Main.o )
/nix/store/hkb0q7w88jkwhmq3lv0q0m0q6agcn612-Main.hi: openBinaryFile: permission denied (Permission denied)

proger commented:
Here we go.
